program/src/state/config.rs

```diff
@@ -0,0 +1,52 @@
+use bytemuck::{Pod, Zeroable};
+use shank::ShankAccount;
+use solana_program::pubkey::Pubkey;
+use spl_discriminator::SplDiscriminate;
+use spl_pod::optional_keys::OptionalNonZeroPubkey;
+
+/// Configuration for a staking system.
+#[repr(C)]
+#[derive(Clone, Copy, Pod, ShankAccount, SplDiscriminate, Zeroable)]
+#[discriminator_hash_input("config::account")]
+pub struct Config {
+    /// Account disciminator.
+    ///
+    /// The discriminator is equal to `ArrayDiscriminator:: UNINITIALIZED` when
+    /// the account is empty, and equal to `Config::DISCRIMINATOR` when the account
+    /// is initialized.
+    pub discriminator: [u8; 8],
+
+    /// Authority that can modify any elements in the config.
+    // TODO: does this need to be optional?
```

Comment:
IMO I think authority makes sense to be optional, but I'm not sure about slash authority.

Reply:
@joncinque Should we keep them as optional? Config authority being optional allows us to "freeze" the configuration when set to `None`; not sure we want to "freeze" the slash authority, unless there is a scenario where the config authority can "temporarily" disable slashing by setting the authority to `None`.

Reply:
If we want to reuse this program a simple lockup, we definitely don't want slashing enabled, so I lean toward keeping it optional